aea/helpers/dialogue/base.py

```diff
@@ -437,65 +391,67 @@ def is_empty(self) -> bool:
         """
         return len(self._outgoing_messages) == 0 and len(self._incoming_messages) == 0
 
-    def update(self, message: Message) -> bool:
+    def is_message_by_self(self, message: Message) -> bool:
+        return message.sender == self.agent_address
+
+    def is_message_by_other(self, message: Message) -> bool:
+        return not self.is_message_by_self(message)
+
+    def has_message(self, message: Message) -> bool:
+        retrieved_message = self.get_message(message.message_id)
+        return message == retrieved_message
+
+    def update(self, message: Message) -> None:
         """
         Extend the list of incoming/outgoing messages with 'message', if 'message' belongs to dialogue and is valid.
 
         :param message: a message to be added
         :return: True if message successfully added, false otherwise
         """
-        if not message.is_incoming:
+        if not message.has_sender:
             message.sender = self.agent_address
 
-        self.ensure_counterparty(message)
-
         if not self.is_belonging_to_dialogue(message):
-            return False
-
-        is_extendable = self.is_valid_next_message(message)
-        if is_extendable:
-            if message.is_incoming:
-                self._incoming_messages.extend([message])
-            else:
-                self._outgoing_messages.extend([message])
-        return is_extendable
+            raise InvalidDialogueMessage("message does not belong to this dialogue.")
```

Comment:
Provide some more info like message id and dialogue label so we know what context the error occurs in

Reply:
Done